### PR TITLE
Migrations for Vaults related tables

### PR DESCRIPTION
**Pull request by teeolendo**
Contains only the migrations agreed on for #507

See mini-ERD below for reference. Only change is that user address is not required for claims table. Wasn't sure if to squash now or later but left it this way for ease of reference.

Distribution_ERD

### CODE REVIEW

**Comment by levity** [comment]
> Only change is that user address is not required for claims table. Wasn't sure if to squash now or later but left it this way for ease of reference.

Yeah, actually this is good for another reason: a circle admin can change the address of a user.

**Review by levity — CHANGES_REQUESTED** [review]

File: hasura/migrations/default/1646250716299_create_table_public_vaults/up.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+CREATE TABLE "public"."vaults" ("id" bigserial NOT NULL, "token_address" varchar, "simple_token_address" varchar, "type" integer NOT NULL, "org_id" bigint NOT NULL, "decimals" integer NOT NULL, "vault_owner" bigint NOT NULL, "created_by" bigint NOT NULL, "created_at" timestamptz NOT NULL DEFAULT now(), "updated_at" timestamptz NOT NULL DEFAULT now(), PRIMARY KEY ("id") , FOREIGN KEY ("org_id") REFERENCES "public"."protocols"("id") ON UPDATE no action ON DELETE no action, FOREIGN KEY ("vault_owner") REFERENCES "public"."users"("id") ON UPDATE no action ON DELETE no action, UNIQUE ("id"));
```
[the remark this answers is not in the document]
Comment (teeolendo) [reply hasura/migrations/default/1646250716299_create_table_public_vaults/up.sql]: Updated accordingly.

File: hasura/migrations/default/1646250716299_create_table_public_vaults/up.sql [hunk 2]
```diff
@@ -0,0 +1,17 @@
+CREATE TABLE "public"."vaults"
+(
+  "id" bigserial NOT NULL,
+  "token_address" varchar,
+  "simple_token_address" varchar,
+  "type" integer NOT NULL,
+  "org_id" bigint NOT NULL,
+  "decimals" integer NOT NULL,
+  "vault_owner" bigint NOT NULL,
```
Comment (levity) [inline hasura/migrations/default/1646250716299_create_table_public_vaults/up.sql:9]:
```suggestion
  "owner_id" bigint NOT NULL,
```

as a visual reminder that it's a foreign key... or `owned_by`? i'm ok with either convention

File: hasura/migrations/default/1646251192439_create_table_public_vault_transactions/up.sql [hunk 3]
```diff
@@ -0,0 +1,40 @@
+CREATE TABLE "public"."vault_transactions"
+(
+  "id" bigserial NOT NULL,
+  "name" varchar NOT NULL,
+  "date_type" varchar NOT NULL,
+  "posneg" varchar NOT NULL,
+  "trx_number" varchar NOT NULL,
```
Comment (levity) [inline hasura/migrations/default/1646251192439_create_table_public_vault_transactions/up.sql:7]:
```suggestion
  "tx_hash" varchar NOT NULL,
```

**Comment by levity** [comment]
Thanks for reformatting the migrations! I'm seeing a few more things now, will review again.

**Comment by teeolendo** [comment]
> Thanks for reformatting the migrations! I'm seeing a few more things now, will review again.

@spydermonkee I've unlocked a new level.

File: hasura/migrations/default/1646251192439_create_table_public_vault_transactions/up.sql [hunk 4]
```diff
@@ -0,0 +1,40 @@
+CREATE TABLE "public"."vault_transactions"
```
Comment (levity) [inline hasura/migrations/default/1646251192439_create_table_public_vault_transactions/up.sql:1]: maybe we can drop `date_type`, `posneg` and `active_users`, and make `value` a numeric type? and then add a column for details/description

File: hasura/migrations/default/1646251192439_create_table_public_vault_transactions/up.sql [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  "id" bigserial NOT NULL,
+  "name" varchar NOT NULL,
+  "date_type" varchar NOT NULL,
+  "posneg" varchar NOT NULL,
+  "trx_number" varchar NOT NULL,
+  "active_users" varchar NOT NULL,
+  "value" varchar NOT NULL,
+  "date" Date NOT NULL,
+  "vault_id" bigint NOT NULL,
+  "created_by" bigint NOT NULL,
```
Comment (levity) [inline hasura/migrations/default/1646251192439_create_table_public_vault_transactions/up.sql:12]:
```suggestion
  "created_by" bigint,
```

we might have vault transactions created from outside the system

File: hasura/migrations/default/1646252376870_create_table_public_claims/up.sql [hunk 6]
```diff
@@ -0,0 +1 @@
+CREATE TABLE "public"."claims" ("id" bigserial NOT NULL, "distribution_id" bigint NOT NULL, "index" bigint NOT NULL, "proof" varchar NOT NULL, "flag" boolean NOT NULL, "amount" bigint NOT NULL, "user_id" bigint NOT NULL, PRIMARY KEY ("id") , FOREIGN KEY ("distribution_id") REFERENCES "public"."distributions"("id") ON UPDATE no action ON DELETE no action, FOREIGN KEY ("user_id") REFERENCES "public"."users"("id") ON UPDATE no action ON DELETE no action, UNIQUE ("id"));
```
Comment (levity) [inline hasura/migrations/default/1646252376870_create_table_public_claims/up.sql:1]: did you want to add the `address` column back in here?

**Review by levity — CHANGES_REQUESTED** [review]

File: hasura/migrations/default/1646251192439_create_table_public_vault_transactions/up.sql (hunk 4 above)
Comment (teeolendo) [reply hasura/migrations/default/1646251192439_create_table_public_vault_transactions/up.sql:1]: Ok, sounds good . what would we store in details/description?

File: hasura/migrations/default/1646252376870_create_table_public_claims/up.sql (hunk 6 above)
Comment (teeolendo) [reply hasura/migrations/default/1646252376870_create_table_public_claims/up.sql:1]: Oh my, yes I did, I can add that on my end.

Comment (teeolendo) [reply hasura/migrations/default/1646252376870_create_table_public_claims/up.sql]: Done

File: hasura/migrations/default/1646251192439_create_table_public_vault_transactions/up.sql (hunk 4 above)
Comment (levity) [reply hasura/migrations/default/1646251192439_create_table_public_vault_transactions/up.sql:1]: It could be for the Details column in the mockup

**Review by levity — APPROVED** [review]

**Pull request merged**